### PR TITLE
fix: make zoomToCursor camera mode to properly work with asynchronous picking 

### PR DESCRIPTION
**Pull request by Savokr**
# Description

Fixes regression introduced by #2191 where `zoomToCursor` camera controls mode was behaving in a wrong way when user initiates a new wheel event. 

# Checklist:

- [x] I am proud of this feature.
- [x] I have performed a self-review of my own code.
- [x] I have manually tested this for different scenarios (different models, formats, devices, browsers).
- [x] I have commented my code in hard-to-understand areas.
- [x] I have made corresponding changes to the public documentation.
- [x] I have added unit and visuals tests to prove that my feature works to the best of my ability.
- [x] I have refactored the code for readability to the best of my ability.
- [x] I have checked that my changes do not introduce regressions in the public documentation.
- [x] I have outlined any known defects / lacking capabilities in the contents of this PR.
- [x] I have listed any remaining work that should follow this PR in the description or jira/miro/etc.
- [x] I have added TSDoc to any public facing changes.
- [x] I have added "developer documentation" in any package README.md that is related / applicable to this PR.
- [x] I have noted down and am currently tracking any technical debt introduced in this PR.
- [x] I have thoroughly thought about the architecture of this implementation.
- [x] I have asked peers to test this feature.

### CODE REVIEW

**Review by larsmoa — CHANGES_REQUESTED** [review]
Eye-bleach! See code comments ;)

File: viewer/packages/camera-manager/src/DefaultCameraManager.ts [hunk 1]
```diff
@@ -554,7 +556,13 @@ export class DefaultCameraManager implements CameraManager {
 
       if (wantNewScrollTarget && isZoomToCursor) {
         scrollStarted = true;
+
+        // Disable controls to prevent camera from moving while picking is happening.
+        // await is not working as expected because event itself is not awaited.
+        this._controls.temporaryDisable(false);
         const newTarget = await this.calculateNewTarget(e);
+        this._controls.temporaryDisable(this._controls.enabledCopy);
```
[the remark this answers is not in the document]
Comment (Savokr) [reply viewer/packages/camera-manager/src/DefaultCameraManager.ts:564]: This doesn't work in all cases, unfortunately. It was exactly the same before without any modifications for ComboControls as well. All this mess is required because two wheel events are issued at the same time and when one event disabled controls, another one reads it and then returns the state of controls to disabled.

File: examples/src/pages/Migration.tsx [hunk 2]
```diff
@@ -116,7 +116,7 @@ export function Migration() {
       (window as any).viewer = viewer;
 
       const controlsOptions: CameraControlsOptions = {
-        changeCameraTargetOnClick: false,
```
[the remark this answers is not in the document]
Comment (Savokr) [reply examples/src/pages/Migration.tsx:119]: It was default to true but changed in Pramods PR, I prefer to have it true by default for easier navigation in the model.

File: viewer/packages/camera-manager/src/ComboControls.ts [hunk 3]
```diff
@@ -144,6 +145,19 @@ export class ComboControls extends EventDispatcher {
...
+  }
+
+  set enabled(value: boolean) {
+    this._enabled = value;
+    this.enabledCopy = value;
+  }
+
+  get enabled(): boolean {
+    return this._enabled;
+  }
```
[the remark this answers is not in the document]
Comment (Savokr) [reply viewer/packages/camera-manager/src/ComboControls.ts:159]: See comment below, this is required because I want to store "ground truth" enabled and return to it after a few disables for controls. It can be done cleaner by just having another variable called "temporary disable" that is totally independent of controls.enable.

File: viewer/packages/webpack.config.js [hunk 4]
```diff
@@ -100,7 +100,7 @@ module.exports = env => {
       ]
     },
 
-    devtool: 'inline-source-map',
```
[the remark this answers is not in the document]
Comment (Savokr) [reply viewer/packages/webpack.config.js:103]: Because I changed sourcemaps to eval everywhere else. This solved the problem of not working source maps.

File: viewer/packages/camera-manager/src/DefaultCameraManager.ts (hunk 1 above)
[the remark this answers is not in the document]
Comment (larsmoa) [reply viewer/packages/camera-manager/src/DefaultCameraManager.ts]: Ok, I see. I think a better fix is to temporarily stop listening to new events, rather than keeping state. This should be done by unregistering from the event listener whenever onWheel fires, and listening again when it finishes. Do it in a `try { ... } finally { ... }` block

File: viewer/packages/camera-manager/src/DefaultCameraManager.ts [hunk 5]
```diff
@@ -545,6 +545,8 @@ export class DefaultCameraManager implements CameraManager {
     };
 
     const onWheel = async (e: any) => {
```
Comment (larsmoa) [inline viewer/packages/camera-manager/src/DefaultCameraManager.ts:547]: Just noticed this `any`. That should be changed asap - there is no need for it....

File: viewer/packages/camera-manager/src/DefaultCameraManager.ts (hunk 1 above)
[the remark this answers is not in the document]
Comment (larsmoa) [reply viewer/packages/camera-manager/src/DefaultCameraManager.ts]: It can probably be done in the same place as here actually.

[the remark this answers is not in the document]
Comment (Savokr) [reply viewer/packages/camera-manager/src/DefaultCameraManager.ts]: It also is not working as expected because it causes "lag" for user interaction. Tried doing that with just a flag for cancelling events that are coming at the same time. Pushed new version now so that you can take a look.

File: viewer/packages/camera-manager/src/DefaultCameraManager.ts (hunk 5 above)
Comment (Savokr) [reply viewer/packages/camera-manager/src/DefaultCameraManager.ts:547]: Can't be easily resolved because of `clickOrTouchEventOffset` : https://github.com/cognitedata/reveal/pull/1307 .
If you can explain why we need it, I can try to remove it and use normal `MouseEvent`.

Comment (larsmoa) [reply viewer/packages/camera-manager/src/DefaultCameraManager.ts:547]: based on what I read the type will simply be `ev: MouseEvent | TouchEvent`

**Review by larsmoa — CHANGES_REQUESTED** [review]
The `controls.enabled` flag is ONLY settable from within `DefaultCameraManager`. Hence, there is no need to add new state in `controls`. This class has _all_ knowledge necessary to handle this here.

Why is this important?
- How would another user of the controls class relate to `temporaryEnabled`?
- How would we replace the current controls in `DefaultCameraManager` with something else? We would need to add the same flag to the new controls which should be unnecessary

Please try to isolate all changes to DefaultCameraManager. If you are not successful, raise your hand.

File: viewer/packages/camera-manager/src/DefaultCameraManager.ts (hunk 5 above)
Comment (Savokr) [reply viewer/packages/camera-manager/src/DefaultCameraManager.ts:549]: It's not that simple unfortunately. `InputHandler` fires "event" with data after applying `clickOrTouchEventOffset` to `MouseEvent`. I can try to create a mock event with new data for proper typing since all fields on normal events are read only.

Comment (Savokr) [reply viewer/packages/camera-manager/src/DefaultCameraManager.ts:549]: Sorry, seems that I was looking at something else, works fine with a few small changes...

**Review by larsmoa — APPROVED** [review]
LGTM, but see one comment

File: examples/webpack.config.js [hunk 6]
```diff
@@ -49,7 +49,7 @@ module.exports = {
   ],
   resolve: {
     extensions: ['.tsx', '.ts', '.js', '.css'],
-    symlinks: true
+    symlinks: false
```
Comment (larsmoa) [inline examples/webpack.config.js:52]: Why is this changed here?

Comment (Savokr) [reply examples/webpack.config.js:52]: It's not a change from this PR, just accidentally commited it, now should be correct
